### DISCUSSION TITLE
CTCS for VT

### DISCUSSION THREAD

**Discussion by ghost** [voxeltycoon/issues#339] [General]
Hello everyone. Today I've decided to realize something that some of you requested: a centralized system to control railway traffic. So I tried to unite all your ideas into a single puzzle that I'll try to explain here.

( ! = this is a not finalized project, it's just an idea so I hope you can help me to improve it)

**Central Traffic Control System [CTCS]:**

_First goal_ = abolition of the green light as default setting for signals.
[condition of use: addition of "scheduled arrival times"].
Signals remain red if no train is scheduled to arrive at that time. When an unplanned train arrives, signals remain red unless a "permit" (give by the player) that allows them to reach their destination is applied to the convoy. Once arrived, the "permit" is self-eliminated after the train has started to follow its schedule. If a train travels late, it will not be blocked by the red signals because these calculate the accumulated time and communicate the go-ahead.

_Second goal_ = improvements to the train schedule.
[condition of use: addition of "scheduled arrival times"].
[condition of use: addition of "day/night cycle"].
Right now into the schedule we have the possibility of choosing station, platform, units to load or unload. From here we can decide arrival times, preferential stop and transit tracks, checkpoints (waypoint).
**1)** Arrival times: such as signals, they allow us having an accurate management of the railway traffic. The arrival time is shown near the name station and next to the timetable there is also an indicator that indicates the time lost, therefore the delay (depending on the amount of delay you can decide whether to skip stops or divert the route).
Furthermore it could be added the possibility to set a time after which the trains can no longer circulate (for example passenger trains stop circulating after 00:00 and return to work after 6:00 a.m.). This feature could make the game a little more difficult for those players at high levels, as well as making it more realistic.
**2)** Tracks: it is essential to distinguish between the main line and the secondary lines, especially in large stations. This is a function similar to the choice of track (which will replace) but with the difference that now you can decide which tracks to prefer for downloading or loading goods. (for example I want to run the express trains on the external tracks, while the freight trains on one of the internal tracks). The advantage is that now you can choose a set of tracks and order them on a scale, from the best to allow trains to pass or stop, at worst.
**3)** Checkpoint (waypoints): checkpoints are a function similar to that performed by the "pass by", but these can be positioned at any point on the railway line, without constraints (except in the vicinity of junctions and intersections). They then determine the route the train must take, which will no longer have to calculate it for itself.

_Third goal_ = CTCS control screen:
[condition of use: addition of "scheduled arrival times"].
[condition of use: addition of "vehicle-infrastructure obsolescence"].
[condition of use: addition of "checkpoints for vehicle routes"].
[condition of use: addition of "accidents and disastrous events"].
[condition of use: addition of "changing junctions direction"].
[condition of use: addition of "tracks status"].
[condition of use: addition of "rail crossing"].
Beyond the standard interface of the game and the white one for the positioning of the signals, a third screen must be introduced from where it is possible to manage all the railway traffic in its various aspects:
- route analysis;
- analysis of the conditions of the trains (locomotives and wagons);
- analysis of infrastructure conditions;
- analysis of warnings.

- modification of the route;
- signal modification;
- modification of the junctions direction;
- modification of rail crossings;
- modification of track status (in operation / closed to traffic, in maintenance, maximum speed).

{note: all the aspects can be easily reached into the "menù" window, and next to each option-line there is a space where a notification will be sometimes added to signaling news about the argument of the line}

**1)** Route analysis: in this option you can see a stylized mini-map of the world with the various vehicle routes, the checkpoints of each route, the vehicles (with their name above, which can be clicked to view their info), signals and their status (green, yellow, red, malfunction), stations (on their name you can click to see their info).
After clicking on the name of vehicles or stations, a table will open in which the various info about them will be listed (arrival times and delays, arrival tracks/stations, units to be loaded or unloaded ...)

**2)** Analysis of the conditions of trains: in this option you can see the wear status of a train. From red (maximum wear) to green (excellent condition) the various wagons and locomotives are colored, whose efficiency decreases if they are in poor condition.
With a high level of wear you go against: reduction of maximum speed, lower transportation capacity, reduced power, higher fuel consumption. Therefore, from this option it is possible to understand which trains to send to the depot for maintaining them, paying as much as the degree of deterioration is high.

**3)** Analysis of infrastructure conditions: in this option you can understand the level of deterioration of bridges, tunnels, tracks and junctions, signals ...
Bridges and tunnel could collapse, deteriored tracks could decrease the trains speed, junctions in bad conditions could jam ...
The verification of infrastructures conditions could be useful when the events like natural catastrophes will be introduced, malfunctioning of the mechanisms or of the circuits of the machines and infrastructures, accidents between vehicles ...

{note: a piece of track can be repaired setting its status on "in maintenance", through the "modification of track status" option}.

**4)** Analysis of warnings: in this option you can see the list of warnings for the entire railway network and the warnings themselves on the mini-map. By clicking on these you can see the information relating to the problem and eventually solve it.

**5)** Modification of the route: this option allows you to modify the route of a vehicle by seeing the changes made on the mini-map in real time. You can change a lot of aspects: checkpoints, arrival time, station and arrival track, how many units must be loaded or unloaded, prevent or allow a station to unload certain goods in certain warehouses, prevent the arrival of certain goods by trucks but allow this by trains and vice versa ..., checkpoints along the line (you can reach the information tables of vehicles and stations by clicking on their name).

**6)** Signal modification: this option can also be reached via the "route analysis" option. There you can also see the individual signals and, by clicking on them, their status (green, yellow, red, malfunctioning) and if necessary change it.
Moreover, from here you can program the signals: they can analyze the speed of the trains (example: divert them to secondary tracks if their speed is low, or on the high-speed track if it is high); they're also able to analyze the cargo (passengers, logs, cement ...) and divert convoys on specific tracks; they can divert the route every a certain number of trains that have passed the signal to reduce traffic (example: trains "1" and "2" go to the right track, trains "3" and "4" to the left track, trains "5" and "6" on the right track, trains numbers "7" and "8" on the left track ...); they are able to analyze the length to divert the longest trains in the longest tracks; they can analyze the fuel.

**7)** Modification of the junctions direction: in this option, which can also be reached through the "route analysis" option, you can divert junctions as you wish.

**8)** Modification of rail crossings: this option, which can also be reached through the "route analysis" option, allows you to set how long before the bars of a rail crossing must get down. The aim of this function is to avoid accidents, to reduce waiting times ... (because every road and rail network has a different traffic density).

**9)** Modification of track status: thanks to this option it's possible to change the status of a track from "in operation" to "closed to traffic" to "in maintenance"; you can also set the speed limit for that section (after that, two small signals will appear near the tracks, at the beginning and at end, indicating the maximum speed).
Into the mini-map, that will appear after choosing this option, paths will be of different colors to indicate the various states of the railway lines.

-------------------------------------------------------------------------------------------------------------

I tried to take into account all your requests made over time, and I hope that the result will satisfy you, given that I have worked enough time with it. :)

**Comment by victorleblais** [comment]
I'm not a dev, but this sounds to me like an almost entirely different game.

Maybe when the game will support "deep" modding, this sort of thing can be done, but I highly doubt the developers would implement such a radically different gameplay in the base game.
